src/components/ClapButton.jsx

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-undef */
+import React, { Component } from 'react';
+import PropType from 'prop-types';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { debounce } from 'lodash';
+import { withRouter } from 'react-router-dom';
+import clapIcon from '../images/clap.svg';
+import * as articleSelector from '../selectors/singleArticleSelector';
+import { getAuthId, getIsAuthenticated } from '../selectors/authSelector';
+import { getToken } from '../selectors/profileSelector';
+import request from '../utils/request';
+import { clapArticleRequest } from '../actions/clapsAction';
+
+/**
+ * A class representing clap button component
+ *
+ * @class ClapButton
+ * @extends {Component}
+ */
+export class ClapButton extends Component {
+  /**
+   * Handles clap request (debounced)
+   *
+   * @returns {void}
+   * @memberof ClapButton
+   */
+  makeClapRequest = debounce(() => {
+    const { clapArticle, articleSlug: slug, token } = this.props;
+    const { clapsCount: claps } = this.state;
+    this.setState({ clapsCount: 0 });
+    return clapArticle({ slug, claps, token });
+  }, 1000);
+
+  static propTypes = {
+    articleSlug: PropType.string.isRequired,
+    articleAuthorId: PropType.number.isRequired,
+    articleClaps: PropType.number.isRequired,
+    clapArticle: PropType.func.isRequired,
+    isLoggedIn: PropType.bool.isRequired,
+    userId: PropType.number,
+    history: PropType.object.isRequired,
+    token: PropType.string.isRequired,
+  };
+
+  static defaultProps = {
+    userId: null,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      clapsLeft: 100,
+      articleClaps: props.articleClaps,
+      clapsCount: 0, // record clap button clicks
+    };
+  }
+
+  componentDidMount = async () => {
+    const {
+      articleSlug: slug,
+      isLoggedIn,
+      userId,
+    } = this.props;
+    if (isLoggedIn) {
+      const response = await request({ route: `articles/${slug}/claps/${userId}` });
```

Comment:
#### Critical
- Not acceptable !!! We already have a pattern we follow with redux.

Reply:
Thanks for the feedback. I have updated the request handling with redux to maintain a single source of truth.